src/connectionInfo/connectionManager.ts

```diff
@@ -0,0 +1,159 @@
+import { isIPv4 } from 'net';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+import { DEFAULT_HOST } from '../utils/constants';
+import { Disposable } from '../utils/dispose';
+import { SETTINGS_SECTION_ID, SettingUtil } from '../utils/settingsUtil';
+import { Connection, ConnectionInfo } from './connection';
+
+/**
+ * @description keeps track of all of the Connection objects and the info needed to create them (ie: initial ports).
+ */
+const localize = nls.loadMessageBundle();
+export class ConnectionManager extends Disposable {
+	private _initHttpPort;
+	private _initWSPort;
+	private _initHost: string;
+	private _connections: Map<vscode.Uri | undefined, Connection>; // undefined key means no workspace root
+
+	constructor(private readonly _extensionUri: vscode.Uri) {
+		super();
+
+		this._initHttpPort = SettingUtil.GetConfig(this._extensionUri).portNumber;
```

Comment:
Are you sure you need the extensionUri to look up configuration? Not sure this is right

Reply:
Hmm yeah, I'm not sure why I thought that was how it needed to be. Will change.